benches/codecs/newline_bytes.rs

```diff
@@ -0,0 +1,90 @@
+use bytes::BytesMut;
+use criterion::{
+    criterion_group, measurement::WallTime, BatchSize, BenchmarkGroup, BenchmarkId, Criterion,
+    SamplingMode, Throughput,
+};
+use std::{fmt, time::Duration};
+use tokio_util::codec::Decoder;
+use vector::codecs::{self, BytesDeserializer, NewlineDelimitedDecoder};
+
+#[derive(Debug)]
+struct Param {
+    slug: &'static str,
+    input: BytesMut,
+    max_length: Option<usize>,
+}
+
+impl fmt::Display for Param {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.slug)
+    }
+}
+
+fn decoding(c: &mut Criterion) {
+    let mut group: BenchmarkGroup<WallTime> =
+        c.benchmark_group("vector::codecs::decoding::Decoder");
+    group.sampling_mode(SamplingMode::Auto);
+
+    for param in &[
+        Param {
+            slug: "no_max",
+            input: BytesMut::from(include_str!("moby_dick.txt")),
+            max_length: None,
+        },
+        Param {
+            slug: "small_max",
+            input: BytesMut::from(include_str!("moby_dick.txt")),
+            max_length: Some(10),
```

Comment:
Is this a reasonable value for `max_length`? That is, are we benchmarking something useful with this setting, or is this just to show a large effect?

Reply:
It's just to show a difference. That there is _such_ a wide divergence between the no-max and small-ish max is what I was getting at; the 10 is an arbitrary smallish value only.